### PR TITLE
enable builds for opensuse-leap-15 #patch

### PR DESCRIPTION
**Pull request by zhmurko**
### Description

[Describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by zhmurko** [comment]
@MarkGibbons could you please take a look into failed build `opensuse-leap-15-sources`. I'm faced a strange bug when `npm_list` helper can not parse empty response from shell command and fails on dependencies list.

**Comment by MarkGibbons** [comment]
As to the error you encountered.  

 begin
        JSON.parse(cmd.run_command.stdout, max_nesting: false)
      rescue JSON::ParserError => e
        Chef::Log.error("nodejs::library::nodejs_helper::npm_list exception #{e}")
        false
      end

The npm list command is going through this branch and returning false.  We'll need to figure out something that validates the returned list before assuming the list is parsed into dependencies.  Possibly by returning [] instead of false.

**Comment by zhmurko** [comment]
temporarry disabled dokken 'opensuse-leap-15-sources' suite
install npm packages  still fails with exit code 7 and message:
```
---- Begin output of npm install  -global express ----
STDOUT: 
STDERR: Error: setuid user id does not exist
    at /usr/local/lib/node_modules/npm/node_modules/uid-number/uid-number.js:49:16
```

**Review by MarkGibbons — APPROVED** [review]
Looks nice.  I'll add a Changelog entry.

File: attributes/packages.rb [hunk 1]
```diff
@@ -4,12 +4,12 @@
 default['nodejs']['packages'] = value_for_platform_family(
   'debian' => node['nodejs']['install_repo'] ? ['nodejs'] : %w(nodejs npm nodejs-dev),
   %w(rhel fedora amazon) => node['nodejs']['install_repo'] ? %w(nodejs nodejs-devel) : %w(nodejs npm nodejs-dev),
-  'suse' => node['platform_version'].to_i < 42 ? %w(nodejs nodejs-devel) : %w(nodejs4 npm4 nodejs4-devel),
+  'suse' => node['platform_version'].to_i < 42 ? %w(nodejs10 npm10 nodejs10-devel) : %w(nodejs4 npm4 nodejs4-devel),
```
Comment (tas50) [inline attributes/packages.rb:7]: We can probably nuke all the < 42 logic now that opensuse 42 is long EOL

Comment (MarkGibbons) [reply attributes/packages.rb:7]: Yanked that from the implementation PR.

**Pull request merged**